src/transformState.js

```diff
@@ -5,7 +5,27 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const obj of actions) {
+    if (obj.type === 'addProperties') {
+      const copyExtra = { ...obj.extraData };
+
+      for (const prop in copyExtra) {
+        state[prop] = copyExtra[prop];
+      }
```

Comment:
This construction works, but use Object.assign(state, obj.extraData); instead. It's shorter, so it makes your code prettyer, more readable and more understandable.